uast/transformer/legacy.go

```diff
@@ -98,19 +100,15 @@ func (n ObjectToNode) transformer() Transformer {
 	)
 }
 
-var _ Transformer = ObjectToNode{}
-
-type RolesDedup struct{}
-
-func (RolesDedup) Do(root uast.Node) (uast.Node, error) {
-	nroot, ok := uast.Apply(root, func(n uast.Node) (uast.Node, bool) {
+func RolesDedup() TransformFunc {
```

Comment:
These are the kind of changes that not only probably optimize everything but also makes the code easier to understand reducing layers of complexity and programmers' cognitive load. :+1: